snapcraft/plugins/copy.py

```diff
@@ -40,10 +42,24 @@ def schema(cls):
 
     def build(self):
         for src in sorted(self.options.files):
-            dst = self.options.files[src]
+            dst = os.path.join(self.installdir, self.options.files[src])
+
             if not os.path.lexists(src):
-                raise EnvironmentError('file "{}" missing'.format(src))
-            dst = os.path.join(self.installdir, dst)
-            os.makedirs(os.path.dirname(dst), exist_ok=True)
-            self.run(['cp', '--preserve=all', '-R', src, dst],
-                     cwd=os.getcwd())
+                # If it doesn't exist, check if it is a glob
+                filelst = glob.glob(src)
+                if not all(os.path.lexists(i) for i in filelst) or \
+                        len(filelst) is 0:
+                    raise EnvironmentError('file "{}" missing'.format(src))
```

Comment:
wouldn't that result in “file "foo/bar*" missing”, which might lead people to think you don't support globs?